govcd/metadata_test.go

```diff
@@ -381,3 +381,476 @@ func (vcd *TestVCD) Test_AddMetadataOnMediaRecord(check *C) {
+
+	// Check if metadata was added correctly
+	metadata, err := vcd.vapp.GetMetadata()
+	check.Assert(err, IsNil)
+	check.Assert(len(metadata.MetadataEntry), Equals, 1)
+	check.Assert(metadata.MetadataEntry[0].Key, Equals, "key")
+	check.Assert(metadata.MetadataEntry[0].TypedValue.Value, Equals, "value")
+}
+
+func (vcd *TestVCD) Test_DeleteMetadataEntryOnVapp(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vApp was not successfully created at setup")
+	}
+	// Add metadata
+	err := vcd.vapp.AddMetadataEntry(types.MetadataStringValue, "key2", "value2")
+	check.Assert(err, IsNil)
+
+	// Remove metadata
+	err = vcd.vapp.DeleteMetadataEntry("key2")
+	check.Assert(err, IsNil)
+
+	metadata, err := vcd.vapp.GetMetadata()
+	check.Assert(err, IsNil)
+	for _, k := range metadata.MetadataEntry {
+		if k.Key == "key2" {
+			check.Errorf("metadata.MetadataEntry should not contain key: 'key2'")
+		}
+	}
+}
+
+func (vcd *TestVCD) Test_AddMetadataEntryOnVm(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+
+	// Find VApp
+	if vcd.vapp.VApp == nil {
+		check.Skip("skipping test because no vApp is found")
+	}
+
+	vapp := vcd.findFirstVapp()
+	vmType, vmName := vcd.findFirstVm(vapp)
+	if vmName == "" {
+		check.Skip("skipping test because no VM is found")
+	}
+
+	vm := NewVM(&vcd.client.Client)
+	vm.VM = &vmType
+
+	// Add metadata
+	err := vm.AddMetadataEntry(types.MetadataStringValue, "key", "value")
+	check.Assert(err, IsNil)
+
+	// Check if metadata was added correctly
+	metadata, err := vm.GetMetadata()
+	check.Assert(err, IsNil)
+	check.Assert(len(metadata.MetadataEntry), Equals, 1)
+	check.Assert(metadata.MetadataEntry[0].Key, Equals, "key")
+	check.Assert(metadata.MetadataEntry[0].TypedValue.Value, Equals, "value")
+}
+
+func (vcd *TestVCD) Test_DeleteMetadataEntryOnVm(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+
+	// Find VApp
+	if vcd.vapp.VApp == nil {
+		check.Skip("skipping test because no vApp is found")
+	}
+
+	vapp := vcd.findFirstVapp()
+	vmType, vmName := vcd.findFirstVm(vapp)
+	if vmName == "" {
+		check.Skip("skipping test because no VM is found")
+	}
+
+	vm := NewVM(&vcd.client.Client)
+	vm.VM = &vmType
+
+	// Add metadata
+	err := vm.AddMetadataEntry(types.MetadataStringValue, "key2", "value2")
+	check.Assert(err, IsNil)
+
+	// Remove metadata
+	err = vm.DeleteMetadataEntry("key2")
+	check.Assert(err, IsNil)
+
+	metadata, err := vm.GetMetadata()
+	check.Assert(err, IsNil)
+	for _, k := range metadata.MetadataEntry {
+		if k.Key == "key2" {
+			check.Errorf("metadata.MetadataEntry should not contain key: 'key2'")
+		}
+	}
+}
+
+func (vcd *TestVCD) Test_DeleteMetadataEntryOnVAppTemplate(check *C) {
```

Comment:
I don't see big difference between tests, maybe just have consistency here and have CRUD one only?

Reply:
Already put together all Add and Delete tests into CRUD ones for simplicity